prompting-client/src/snapd_client/interfaces/mod.rs

```diff
@@ -205,12 +212,12 @@ impl TryFrom<RawPrompt> for TypedPrompt {
         // SAFETY: we are only deserializing the prompt constraints data after checking the value
         //         of raw.interface is correct for the SnapInterface we are using.
         unsafe {
-            if raw.interface == HomeInterface::NAME {
-                Ok(TypedPrompt::Home(Prompt::try_from_raw(raw)?))
-            } else {
-                Err(Error::UnsupportedInterface {
+            match raw.interface.as_str() {
+                HomeInterface::NAME => Ok(TypedPrompt::Home(Prompt::try_from_raw(raw)?)),
+                CameraInterface::NAME => Ok(TypedPrompt::Camera(Prompt::try_from_raw(raw)?)),
+                _ => Err(Error::UnsupportedInterface {
```

Comment:
ok, so that trace back what I wrote about the `UnsupportedInterface`. I let you trace back what we do in that case when getting this error.